apis/kueue/v1beta1/workload_types.go

```diff
@@ -336,6 +358,11 @@ const (
 // +kubebuilder:resource:shortName={wl}
 
 // Workload is the Schema for the workloads API
+// +kubebuilder:validation:XValidation:rule="has(self.spec) && has(self.status) && has(self.status.conditions) && self.status.conditions.exists(c, c.type == 'QuotaReserved' && c.status == 'True') && has(self.status.admission) ? size(self.spec.podSets) == size(self.status.admission.podSetAssignments) : true", message="podSetAssignments must have the same number of podSets as the spec"
+// +kubebuilder:validation:XValidation:rule="(has(oldSelf.spec) && has(oldSelf.status) && has(oldSelf.status.conditions) && oldSelf.status.conditions.exists(c, c.type == 'QuotaReserved' && c.status == 'True')) ? (oldSelf.spec.podSets == self.spec.podSets) : true", message="field is immutable"
```

Comment:
Oh this check in particular will also be problematic for #1852

Please keep this one in the webhook for now.

Reply:
Done